addons/func_godot/src/map/func_godot_map.gd

```diff
@@ -609,6 +627,18 @@ func build_entity_collision_shapes() -> void:
 			
 			var collision_shape: CollisionShape3D = entity_collision_shapes[entity_idx][0]
 			collision_shape.set_shape(shape)
+			
+			if entity_definition and entity_definition.add_face_shape_index_metadata:
+				# TODO: face shape stuff gets thrown away here, building the array in surface gatherer could be avoided altogether for concave
```

Comment:
Don't throw it away. Some users may want this for some reason or another, use cases aren't limited to what we think of.

Reply:
How can I keep it? The concept of `collision_shape_to_face_range_map` doesnt make sense in this case, since each key corresponds to a single Vector2i, and there is only one possible key: the name of the single collision shape generated for this node.
could be solved by making each node name a key to an Array[Vector2i] instead... a bit odd since the arrays will always contain one item when building convex.

Reply:
I'm not sure why this doesn't make sense. I know that there'll only be one possible key in a concave collision entity, but the point is that the data is consistent between convex and concave. Devs may want or need unified methods between the two. Once again, even if it may not seem useful to you, it might be useful to someone else and there's no reason to change the behavior in this instance between convex and concave.

Don't overthink it or overdesign it too much. If you're already grabbing each collision shape and providing a face range for each, it doesn't matter if it's 10 shapes or 1: the output data should remain consistent. If someone wanted that data on a concave collision entity for some reason, far be it from us to deny them that data for arbitrary reasons.

Reply:
Currently, the output for a concave collision shape is something along the lines of `{ "entity_#_collision_shape_0": Vector2i(0, num_faces) }`. Is this acceptable? If so, I can avoid storing/building information about which brushes correspond to which faces (which is what I do in the case of convex).
When I say I am throwing the data away, I am talking about that brush -> faces mapping, which is internal and only really useful in the convex case. My thought was that maybe the user would still want that data when building concave collision. I was thinking pretty much what you are saying about not denying the user access to data that we already have. But yes it is a bit of overthinking since it would require a change in the format to provide it to them.

Reply:
As long as the output data the user receives is the same format, then skipping calculations when you already know the answer is perfectly acceptable (and preferred). The only 4 things that matter are Output Data, Input Method, Readability, and Performance, preferably in that order. If those are accomplished we're doing it right.